### PR TITLE
docs(grid): Enchancing API documentation #13197

### LINKED ISSUE #13197: Enhance Grid API documentation
For example, properties of the [`IGridEditDoneEventArgs`](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/interfaces/IGridEditDoneEventArgs.html#rowData) lack comments:

All properties should be acoompanied by a comment, like in the [`IPinColumnEventArgs`](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/interfaces/IPinColumnEventArgs.html#rowData)

Enhance the docs for all events listed in https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/grids/common/events.ts that lack documentation

### PR DESCRIPTION
**Pull request by GrblackStar**
Closes #13197

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [x] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/common/enums.ts [hunk 1]
```diff
@@ -1,26 +1,51 @@
 
 import { mkenum } from '../../core/utils';
+/**
+ * Enumeration representing different filter modes for grid filtering.
+ * - quickFilter: default value
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/common/enums.ts:5]:
```suggestion
 * - quickFilter: Default mode with a filter row UI between the column headers and the first row of records.
```

File: projects/igniteui-angular/src/lib/grids/common/enums.ts [hunk 2]
```diff
@@ -29,35 +54,61 @@ export type GridKeydownTargetType =
...
+ * Enumeration representing different selection modes for the grid columns if can be selected.
+ * - none: No selection is allowed.
+ */
 export const GridSelectionMode = mkenum({
     none: 'none',
     single: 'single',
     multiple: 'multiple',
     multipleCascade: 'multipleCascade'
 });
+/** Type alias for the keys of the GridSelectionMode enumeration. */
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/common/enums.ts:67]: This comment should be removed.

File: projects/igniteui-angular/src/lib/grids/common/enums.ts [hunk 3]
```diff
@@ -29,35 +54,61 @@ export type GridKeydownTargetType =
...
 export type GridSelectionMode = (typeof GridSelectionMode)[keyof typeof GridSelectionMode];
 
+/** Enumeration representing different column display order options. */
 export const ColumnDisplayOrder = mkenum({
     Alphabetical: 'Alphabetical',
     DisplayOrder: 'DisplayOrder'
 });
+/** Type alias for the keys of the ColumnDisplayOrder enumeration. */
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/common/enums.ts:75]: This comment should be removed

File: projects/igniteui-angular/src/lib/grids/common/enums.ts [hunk 4]
```diff
@@ -29,35 +54,61 @@ export type GridKeydownTargetType =
...
+ * - Bottom: Rows are pinned to the bottom of the grid.
+ */
 export enum RowPinningPosition {
     Top,
     Bottom
 }
 
+/**
+ * Enumeration representing different paging modes for the grid.
+ * - Local: Default value; The grid will paginate the data source based on the page
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/common/enums.ts:100]:
```suggestion
 * - Local: The grid will use local data to extract pages during paging.
 * - Remote: The grid will expect pages to be delivered from a remote location and will only raise events during paging interactions.
```

File: projects/igniteui-angular/src/lib/grids/common/enums.ts [hunk 5]
```diff
@@ -29,35 +54,61 @@ export type GridKeydownTargetType =
...
+ */
 export enum GridPagingMode {
     Local,
     Remote
 }
 
+/**
+ * Enumeration representing different types of grid instances.
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/common/enums.ts:108]: This should not be documented.

File: projects/igniteui-angular/src/lib/grids/common/enums.ts (hunk 4 above)
Comment (GrblackStar) [reply projects/igniteui-angular/src/lib/grids/common/enums.ts]: Will include in the commit

Comment (kdinev) [reply projects/igniteui-angular/src/lib/grids/common/enums.ts]: @GrblackStar Just FYI, for the suggested comments, you can click "commit suggestion" from the UI. You do not have to replicate it in a different commit.

**Review by kdinev — APPROVED** [review]

**Pull request merged**